crates/runtime/src/instance.rs

```diff
@@ -394,67 +259,65 @@ impl Instance {
     /// Return the indexed `VMTableDefinition`.
     #[allow(dead_code)]
     fn table(&self, index: DefinedTableIndex) -> &VMTableDefinition {
-        table(&self.vmctx, &self.offsets, index)
+        unsafe { &*self.table_ptr(index) }
     }
 
     /// Return the indexed `VMTableDefinition`.
-    #[allow(dead_code)]
-    fn table_mut(&mut self, index: DefinedTableIndex) -> &mut VMTableDefinition {
-        table_mut(&mut self.vmctx, &self.offsets, index)
+    fn table_ptr(&self, index: DefinedTableIndex) -> *mut VMTableDefinition {
+        unsafe { self.tables_ptr().add(index.as_u32() as usize) }
```

Comment:
This changes `usize::try_from(...).unwrap()` with a `u32` operand to `as usize`. I don't know how important or realistic 16-bit platforms are here, but my main concern is that we have `usize::try_from...unwrap()` with a `u32` operand in other similar places in the code, and I'd like to keep things consistent one way or another.

I picked `try_from...unwrap()` originally because I don't like how Rust's `as` is a single syntax for both lossless and several forms of lossy conversions, so it doesn't carry programmer intent. And all the places where we cast indices to `usize` and add them to pointers should be factored out into small helper functions like `table_ptr` here, so it shouldn't add too much clutter to the overall codebase.

But all that said, my bigger concern here is just picking a style and using it consistently.

Reply:
Nah that sounds reasonable to me! I'll switch to `try_from` here instead of using `as`. Agreed that Rust hasn't done a great job of helping projects control `as` usage :(